languages/scala/reference/README.md

```diff
@@ -0,0 +1,193 @@
+# Scala reference
+
+## Concepts
+
+The Scala concept exercises are based on concepts. The list below contains the concepts that have been identified for the Scala language.
+
+### Language-unique
+
+- Pattern matching
+- Functional Programming
+- Immutability
+
+### Functional
+
+- [Functions][functions]
+  - [Pure functions][pure_functions]
+  - [Higher-order functions][higher_order_functions]
+  - [Anonymous functions][anonymous_functions]
+  - [Nested functions][nested_functions]
+  - [Function composition][function_composition]
+  - [Partial application][partial_application]
+  - [Recursion][recursion]
+    - Tail recursion
+  - Currying
+  - Implicit returns
+  - Type annotations
+- [Immutability][immutability]
+- [Pattern matching][pattern_matching]
```

Comment:
I think `unapply` is a valuable concept. I'm just not entirely sure where to put it. I thought of pattern matching since this would be the best use case. Wdyt?

Reply:
Ah, nevermind, just found the `unapply`. Sorry for the confusion.